connectivity/lwipstack/source/LWIPInterface.cpp

```diff
@@ -441,6 +442,11 @@ nsapi_error_t LWIP::add_ethernet_interface(EMAC &emac, bool default_if, OnboardN
     interface->memory_manager = &memory_manager;
     interface->ppp_enabled = false;
 
+    hostname = user_network_interface->get_hostname();
+    if(hostname) {
+        netif_set_hostname(&interface->netif, hostname);
```

Comment:
Does doing this in add_ethernet_interface() mean that you have to call set_hostname() before adding the Ethernet interface?  What about if the user calls set_hostname() later?  We should probably document this in the NetworkInterface comments.

Reply:
Yes, `EthernetInterface::set_hostname` should be called before `EthernetInterface::connect` (which calls `LWIP::add_ethernet_interface`). If it's called later, `EthernetInterface::set_hostname` will return `NSAPI_ERROR_BUSY`, as the interface is already initialized.

This behaves exact like `EthernetInterface::set_mac_address`.

Reply:
We can rewrite NetworkInterface::set_hostname comments to address this behavior:
```
     *  @retval         NSAPI_ERROR_BUSY if interface is already initialized
```

I just think we should change it in both ```NetworkInterface::set_hostname``` and ```NetworkInterface::set_mac_address``` comments for consistency. What do you think?